components/nimbus/src/stateful/enrollment.rs

```diff
@@ -180,6 +180,47 @@ pub fn unenroll_for_pref(
     Ok(events)
 }
 
+pub fn add_previous_state_for_experiment(
```

Comment:
This should be pub(crate) at most, we don't really need access to this outside of the crate.

It should probably also be a static fn next to the `NimbusClient::register_previous_state` method

Reply:
Moved, thanks!